oauth2/handler.go

```diff
@@ -250,6 +250,9 @@ func (h *Handler) AuthHandler(w http.ResponseWriter, r *http.Request, _ httprout
 		h.writeAuthorizeError(w, authorizeRequest, errors.Wrapf(fosite.ErrServerError, "Could not store session cookie: %s", err))
 		return
 	}
+	
+	// add key id to header
+	session.Headers.Add("kid", "public")
```

Comment:
Adding the kid [here](https://github.com/Grillz/hydra/blob/e868ef9e860003e8035bc40da5d02c6fda26fe90/oauth2/consent_strategy.go#L93) should work